iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp

```diff
@@ -64,6 +64,14 @@ struct FusionOfTensorOpsPass
             }
           }
 
+          // Limit the number of operands. We have hard limit (32) of bindings
+          // passing down to HAL. Set the number to be as same as the limit --
+          // IREE_HAL_MODULE_MAX_DESCRIPTOR_BINDING_COUNT.
+          constexpr int64_t kIreeMaxOperandCount = 32;
+          auto numOperands = producer.getOwner()->getNumOperands() +
```

Comment:
Do you mind trying 

```
llvm::DenseSet<Value> operands;
operands.insert(producer->operands_begin(), producer->operands_end())
operands.insert(consumer->operands_begin(), std::next(consumer->operands_begin(), consumer.getOperandNum()));
operands.insert(std::next(consumer->operands_begin() + consumer.getOperandNum() + 1), consumer->operands_end());
if (operands.size() >= kIreeMaxOperandCount) ...
```

basically this accounts for the case where we might end up with duplicate operands with number of operands > 32, but after canonicalization when duplicates are dropped, we get below the limit.
Either case, this is fine.

Reply:
Good catch, let me take a stab at it.

Reply:
hmm, the lambda must take two `const OpOperand &`. However, the OpOperand::getOperandNumber method is not marked as const. So we can not call it in the function.

Reply:
Ugh, thats horrible. Can you try changing the signature of the callback upstream to not use `const OpOperand &`. Just add documentation to say that it is expected that `OpOperand` is not modified. Unfortunately MLIR does not add `const` to methods (and seems like the underlying implementation of that method, calls other methods, so it would be painful to add `const` everywhere)

Reply:
I'll land what we have now, we can revisit it later.

Reply:
Ah, I missed the comment before merging the PR! Let me try it and let's see what's happening.

Reply:
It works with `(const OpResult &producer, OpOperand &consumer)`. Let me send a fix upstream, and fix the IREE side later.